### PR TITLE
[Plugin] MoreSearchEngines

### PR DESCRIPTION
**Pull request by ImpishMoxxie**
## Allows you to search in a few different search engines!

**Only searches text from msgs**

plugin request thingy: https://github.com/Vencord/plugin-requests/issues/117

img

yes we added some funny sites :3

### CODE REVIEW

File: src/plugins/MoreSearchEngines/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                                <img
+                                    style={{
+                                        borderRadius: i >= 3 // Do not round Google, Yandex & SauceNAO
+                                            ? "50%"
+                                            : void 0
+                                    }}
+                                    aria-hidden="true"
+                                    height={16}
+                                    width={16}
+                                    src={new URL("/favicon.ico", Engines[engine]).toString().replace("lens.", "")}
```
Comment (Vendicated) [inline src/plugins/MoreSearchEngines/index.tsx:52]:
```suggestion
                                    src={new URL("/favicon.ico", Engines[engine]).toString()}
```
none of your engines contain lens. (anymore?)

File: src/plugins/MoreSearchEngines/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                        action={() => search(src, Engines[engine])}
+                    />
+                );
+            })}
+        </Menu.MenuItem>
+    );
+}
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => () => {
+    const group = findGroupChildrenByChildId("copy-link", children);
```
Comment (Vendicated) [inline src/plugins/MoreSearchEngines/index.tsx:66]:
```suggestion
	if (!message) return;

    const group = findGroupChildrenByChildId("copy-link", children);
```

**Comment by Vendicated** [comment]
double search... why not replace the native discord option instead?

image

**Comment by Kubuxu** [comment]
I was looking at writing a plugin like this but for a slightly different use path.
I wanted to replace the `Search with Google` with my preferred search engine.

Also, could you add Kagi to the list? `https://kagi.com/search?q=`

**Comment by ImpishMoxxie** [comment]
> I was looking at writing a plugin like this but for a slightly different use path. I wanted to replace the `Search with Google` with my preferred search engine.
> 
> Also, could you add Kagi to the list? `https://kagi.com/search?q=`


we made it so that now it's under the "search with Google" and we are working on making it only search the highlighted part.
also ya I can add kagi.com

**Comment by ImpishMoxxie** [comment]
updated it to show under the "search with Google" as https://github.com/Vendicated/Vencord/pull/2218#issuecomment-1972188521 asked

added settings to select what engines to use

only shows now when marking text and uses the marked text for the search

File: src/plugins/MoreSearchEngines/index.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import PluginModal from "@components/PluginSettings/PluginModal";
+import { Devs } from "@utils/constants";
+import { openModal } from "@utils/modal";
+import definePlugin, { OptionType } from "@utils/types";
+import { Menu, SettingsRouter } from "@webpack/common";
+
+const testieWestie: boolean = true;
+
+const Engines = {
+    DuckDuckGo: "https://duckduckgo.com/",
```
Comment (Vendicated) [inline src/plugins/MoreSearchEngines/index.tsx:19]: where's Google?

Comment (ImpishMoxxie) [reply src/plugins/MoreSearchEngines/index.tsx:19]: google still shows this is just extra browsers

image

**Comment by Vendicated** [comment]
> updated it to show under the "search with Google" as #2218 (comment) asked

that's absolutely not what i asked.. i suggested to **replace** the Google search option

**Pull request closed without merging**